Sources/StreamChat/ChatClient.swift

```diff
@@ -367,27 +364,52 @@ public class ChatClient {
             completion: { completion?($0) }
         )
     }
-    
+
     /// Disconnects the chat client from the chat servers. No further updates from the servers
     /// are received.
+    @available(*, deprecated, message: "Use the asynchronous version of `disconnect` for increased safety")
     public func disconnect() {
-        clientUpdater.disconnect(source: .userInitiated) {
+        disconnect {}
+    }
+
+    /// Disconnects the chat client from the chat servers. No further updates from the servers
+    /// are received.
+    public func disconnect(completion: @escaping () -> Void) {
```

Comment:
I'm still concerned about this API change. Would be nice to have some customers try this new API and see the potential impact on their apps. If this is problematic we could opt in the approach of moving this to the login side.

Reply:
The previous API will still be available, under a deprecation note.